src/main/java/eu/miaplatform/customplugin/springboot/GracefulShutdown.java

```diff
@@ -0,0 +1,22 @@
+package eu.miaplatform.customplugin.springboot;
+
+import org.springframework.beans.factory.annotation.Value;
+import org.springframework.context.ApplicationListener;
+import org.springframework.context.event.ContextClosedEvent;
+import org.springframework.stereotype.Component;
+
+@Component
+public class GracefulShutdown implements ApplicationListener<ContextClosedEvent> {
+
+    @Value("${delay.shutdown.seconds:0}")
```

Comment:
Since the default from the environment variable is 10, let's set the default here to 10 too so we don't get misleaded by the different values.

Reply:
```suggestion
    @Value("${delay.shutdown.seconds:10}")
```